pingcap/tidb
# server: add hot region table/index api

### What problem does this PR solve? 

fixes #7870 

### What is changed and how it works?

add http api to query all hot table/index

### Check List

Tests 

 - Manual test (add detailed scripts or steps below)


1. setup PD/TiKV/TiDB/Promethus/Grafana in local machine
2. use sysbench to prepare data, this will got hot region
3. then acess http://0.0.0.0:10080/regions/hot to see hot table/idx list in desc order

```
{
  "read": [
    
  ],
  "write": [
    {
      "db_name": "sbtest1",
      "table_name": "sbtest13",
      "index_name": "",
      "flow_bytes": 220718,
      "max_hot_degree": 12,
      "region_count": 1
    },
    {
      "db_name": "sbtest1",
      "table_name": "sbtest1",
      "index_name": "",
      "flow_bytes": 134524,
      "max_hot_degree": 11,
      "region_count": 1
    },
    {
      "db_name": "sbtest1",
      "table_name": "sbtest12",
      "index_name": "",
      "flow_bytes": 115753,
      "max_hot_degree": 11,
      "region_count": 1
    },
    {
      "db_name": "sbtest1",
      "table_name": "sbtest4",
      "index_name": "",
      "flow_bytes": 214674,
      "max_hot_degree": 10,
      "region_count": 1
    }
  ]
}
```
then we can using `jq` tool to filter the result

Code changes

 - Has exported http_api and metric change
 - Has exported variable/fields change

Side effects

 - No

Related changes

 - N/A

### Remain question

- code refine


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/7890)
